doc/source/scripts/base.py

```diff
@@ -18,87 +70,73 @@ class TableWriter(object):
     Filename relative to source.
     """
 
-    filename = ""
-    include_table = False
-    headings = []
-    preprocess = []
-    postprocess = []
-    sort = True
+    def __init__(
+        self,
+        filename: str = "",
+        include_table: Optional[str] = None,
+        sort: bool = True,
+        input_items: Iterable[Any] = set(),
+        column_spec: list[tuple[str, Callable[..., str]]] = [],
+        lines: list[list[str]] = [],
+    ):
+        if column_spec:
+            assert input_items
+        assert (column_spec and not lines) or (lines and not column_spec)
+        stem = os.getcwd().split("source")[0]
+        self.path = os.path.join(stem, "source", filename)
+        self.filename = filename
+        self.include_table = include_table
+        self.sort = sort
+        self.input_items = input_items
+        self.column_spec = column_spec
+        self.lines = lines
```

Comment:
```suggestion
        self.column_spec = list(column_spec)
        self.lines = list(lines)
```

I know it's for internal use, but it's always a bit sketchy having lists as default arguments -- just safeguarding it a bit here.

Reply:
Ah, that's such a n00b error – I actually ran flake8 yesterday and found this, forgot to push the commit